### PR TITLE
gh-126313: Fix a crash in `curse.napms` due to incorrect error handling

### PR DESCRIPTION
**Pull request by picnixz**
* Issue: gh-126313

### CODE REVIEW

**Comment by picnixz** [comment]
The backports will need to be manual because they will be a conflict with the new structure of the `curses` module (I haven't changed the macro for this one).

File: Modules/_cursesmodule.c [hunk 1]
```diff
@@ -3815,8 +3815,11 @@ static int
 _curses_napms_impl(PyObject *module, int ms)
 /*[clinic end generated code: output=5f292a6a724491bd input=c6d6e01f2f1df9f7]*/
 {
-    PyCursesStatefulInitialised(module);
-
+    if (!_PyCursesStatefulCheckFunction(module,
```
Comment (rruuaanng) [inline Modules/_cursesmodule.c:3818]: Hmmmm. It looks good, I admit. But the difference is only in `return 0` and `return -1`. It seems a bit cumbersome.

Comment (picnixz) [reply Modules/_cursesmodule.c:3818]: It is necessary. Returning -1 with AC when an exception will re-raise this exception. Returning 0 in the other cases is to allow conversion to a NULL pointer when the return type is a `PyObject *`.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by picnixz** [comment]
(I think I forgot to include the latest CI tasks in this branch, which is why it was stuck)

**Comment by picnixz** [comment]
I won't have time for the backports, so I'll do it upon my return (Wednesday). Anyone else is free to try.

**Comment by picnixz** [comment]
Update: No need for a 3.12 backport (see the issue for details)

**Pull request merged**